mobile-app/app/components/CreateWalletStepIndicatorV2.tsx

```diff
@@ -0,0 +1,146 @@
+import { Platform, StyleProp, ViewStyle } from 'react-native'
+import { Text, View } from '.'
+import { useThemeContext } from '@shared-contexts/ThemeProvider'
+import { tailwind } from '@tailwind'
+import { translate } from '@translations'
+import { ThemedIcon, ThemedTextV2 } from './themed'
+
+interface StepIndicatorProps {
+  current: number
+  total?: number
+  steps?: string[]
+  style?: StyleProp<ViewStyle>
+}
+
+export const CREATE_STEPS = [
+  'RECOVERY',
+  'VERIFY',
+  'SECURE'
+]
+
+export const RESTORE_STEPS = [
+  'RESTORE',
+  'SECURE'
+]
+
+/**
+ * @param props
+ * @param {number} props.total throw error if not fulfill (3 < total < 6), optional when props.steps provided
+ * @param {number} props.current throw error if not fulfill (0 < current <= total)
+ * @param {string[]} props.steps description displayed for each step
+ * @returns {JSX.Element}
+ */
+export function CreateWalletStepIndicatorV2 (props: StepIndicatorProps): JSX.Element {
+  const { isLight } = useThemeContext()
+  const {
+    current,
+    total,
+    style: containerViewStyle,
+    steps = []
+  } = props
+  if (total === undefined && steps.length === 0) {
+    throw Error('Invalid prop for CreateWalletStepIndicator')
+  }
+
+  const totalStep = total ?? steps.length
+  if (totalStep > 5 || current <= 0 || current > totalStep) {
+    throw Error('Invalid prop for CreateWalletStepIndicator')
+  }
+
+  function following (): JSX.Element[] {
+    const arr: JSX.Element[] = []
+    for (let i = 1; i < totalStep; i++) {
+      const lineStyle = current >= i + 1 ? 'bg-green-v2' : isLight ? 'bg-mono-light-v2-900' : 'bg-mono-dark-v2-900'
+      arr.push(
+        <View
+          key={i * 2}
+          style={tailwind(`h-px flex-grow mt-4 ${lineStyle}`)}
+        />)
+      arr.push(
+        <StepNode
+          content={steps[i]}
+          current={current}
+          isLight={isLight}
+          key={i * 2 + 1}
+          step={i + 1}
+        />)
+    }
+    return arr
+  }
+
+  return (
+    <View style={[tailwind('flex-col justify-center items-center w-full'), containerViewStyle]}>
+      <View style={tailwind('flex-row justify-center w-9/12 h-16')}>
```

Comment:
let's increase the width slightly, looks better
```suggestion
      <View style={tailwind('flex-row justify-center w-10/12 h-16')}>
```